### PR TITLE
eks-prow-build-cluster: Use dedicated Managed Node Groups (MNGs) per Availability Zone (AZ)

### PR DESCRIPTION
**Pull request by xmudrii**
xref https://github.com/kubernetes/k8s.io/issues/6303 (see for more details and context)

As per recommendations received from the AWS support and @tzneal, we're replacing blue and green node groups with node groups per AZ. In other words, old node groups had instances in all three AZs. Now, we have a dedicated node group for each AZ. This is a short-term solution to fix the stability issues that we're facing.

This has been successfully rolled out to canary, I'll do prod rollout once this PR is reviewed and merged.

Notes:

- We plan to switch to Karpenter long-term (https://github.com/kubernetes/k8s.io/issues/5168)
- We didn't want to suspend AZRebalacing process because it's not natively supported by EKS, so we opted in for dedicated node groups as that solution can be fully-automated using Terraform
- We have dedicated Terraform objects and variables for each Node Group. I explicitly didn't want to use `count` because that would make rolling out upgrades too complicated

Follow-ups:

- Remove blue/green Terraform variables and objects
- Update docs to reflect changes in the upgrade procedure

/assign @upodroid @ameukam @dims

### CODE REVIEW

**Review by upodroid — APPROVED** [review]
LGTM

**Comment by upodroid** [comment]
cancel the hold when you are ready to merge
/hold

**Comment by xmudrii** [comment]
Let's get this merged and I'll create a new PR for follow-ups
/hold cancel

File: infra/aws/terraform/prow-build-cluster/node_group_us-east-2a.tf [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+locals {
+  node_group_build_us_east_2a = {
+    name            = "build-us-east-2a"
+    description     = "EKS managed node group in US East 2a"
+    use_name_prefix = true
+
+    cluster_version = var.node_group_version_us_east_2a
+
+    taints = var.node_taints_build
```
Comment (bryantbiggs) [inline infra/aws/terraform/prow-build-cluster/node_group_us-east-2a.tf:25]: FYI - you can use [`eks_managed_node_group_defaults`](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/2cb1fac31b0fc2dd6a236b0c0678df75819c5a3b/examples/eks_managed_node_group/main.tf#L84)  to set the default values you wish to use across all of the managed nodegroups, and then any specific settings unique to that nodegroup can be set in the nodegroup definition. You can also set the defaults and still override the value in the nodegroup definition as well

So roughly speaking, something like the following might help cut down on the copy+pasta across nodegroup definitions:

```hcl
  eks_managed_node_group_defaults = {
    use_name_prefix = true
    
    taints = var.node_taints_build
    labels = var.node_labels_build
    ... anything else that you want common across the nodegroups
  }
  
  
```

Comment (xmudrii) [reply infra/aws/terraform/prow-build-cluster/node_group_us-east-2a.tf:25]: That's neat, thanks for pointing it out! I'll keep it in mind and see if we can make use of it the next time we do some rollout

**Pull request merged**